### PR TITLE
Allow reformatting an entire SmalltalkWorkspace if nothing is selected

### PR DESCRIPTION
**Pull request by daniels220**

### CODE REVIEW

**Review by blairmcg — CHANGES_REQUESTED** [review]

File: Core/Object Arts/Dolphin/IDE/Base/SmalltalkWorkspace.cls [hunk 1]
```diff
@@ -1134,16 +1134,14 @@ reformatSource
 	Attempts to maintain the base indentation of the first selected line."
 
 	| source formatter reformatted expr tabs |
+	self hasSelection ifFalse: [self view selectAll].
```
[the remark this answers is not in the document]
Comment (daniels220) [reply Core/Object Arts/Dolphin/IDE/Base/SmalltalkWorkspace.cls]: Eh, it's easy enough, and since I'm fixing the package/categories anyway...

**Comment by blairmcg** [comment]
@daniels220, inserting a `self view moveToEndOfDocument` as the last statement of the false branch of the final if-statement will fix the test, and then this can be merged. Although only ScintillaView implements the `moveToEndOfDocument` message, it's fine to assume that the view of a workspace will always be a ScintillaView as it can no longer be anything else really.

**Comment by blairmcg** [comment]
@daniels220, I'll merge this, and then fix the pre-existing test that is failing, since it is just assuming the old rounding behaviour, rather than the new truncating behaviour.

**Comment by daniels220** [comment]
Sorry, lost track of this. Added the `moveToEndOfDocument` as you suggested.

**Pull request merged**